blob/azureblob/azureblob.go

```diff
@@ -419,11 +420,14 @@ func NewPipeline(credential azblob.Credential, opts azblob.PipelineOptions) pipe
 // write and delete operations on objects within it.
 // See https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction.
 type bucket struct {
-	name         string
-	pageMarkers  map[string]azblob.Marker
-	serviceURL   *azblob.ServiceURL
-	containerURL azblob.ContainerURL
-	opts         *Options
+	name                  string
+	pageMarkers           map[string]azblob.Marker
+	serviceURL            *azblob.ServiceURL
+	containerURL          azblob.ContainerURL
+	opts                  *Options
+	credentialExpiration  *time.Time
```

Comment:
Formatting nits:

* Can you insert a blank line before the 3 new fields?
* Rename mtx -> mu and add an inline comment "// mu protect the fields below."

(This is consistent with how mutexes are used in this repo, and make it clear
what the mutex is used for).

Thanks!